tests/test_main.py

```diff
@@ -14,3 +16,33 @@ def runner() -> CliRunner:
 def test_main_fails(runner: CliRunner) -> None:
     result = runner.invoke(__main__.main)
     assert result.exit_code == 2
+
+
+def test_retrieve_called_with_auth_headers(
+    request: pytest.FixtureRequest, runner: CliRunner
+) -> None:
+    if request.config.getoption("typeguard_packages"):
+        pytest.skip(
+            "The import hook that typeguard uses seems to break the mock below."
+        )
+
+    expected_auth_header = {"Authorization": "api-key fake-api-key-value"}
+    with unittest.mock.patch(
+        "stac_api_validator.validations.retrieve"
+    ) as retrieve_mock:
+        runner.invoke(
+            __main__.main,
+            args=[
+                "--root-url",
+                "https://invalid",
+                "--conformance",
+                "core",
+                "--auth-headers",
+                f"{expected_auth_header}",
+            ],
+        )
+        assert retrieve_mock.call_count == 1
+        r_session = retrieve_mock.call_args.args[-1]
+        assert (
+            r_session.headers["Authorization"] == expected_auth_header["Authorization"]
+        )
```

Comment:
Since most of this PR's changes are to `validations.py`, I'd recommend adding tests that exercise the new behavior for the following functions:

- `get_catalog` (a new "public" function)
- `retrieve`
- `validate_api`
- `validate_browsable`

@philvarner may have a preferred test layout, but in liu of that, they could live in a `test_validations.py`.